lib/lightning/maintenance/partition_table_service.ex

```diff
@@ -0,0 +1,182 @@
+defmodule Lightning.PartitionTableService do
+  @moduledoc """
+  Service to keep the partition tables up to date.
+  """
+
+  use Oban.Worker,
+    queue: :background,
+    max_attempts: 1
+
+  import Ecto.Query
+
+  alias Lightning.Repo
+
+  require Logger
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"add_headroom" => %{"weeks" => weeks}}})
+      when is_integer(weeks) do
+    add_headroom(:all, weeks)
+  end
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"drop_older_than" => %{"weeks" => weeks}}})
```

Comment:
I could not get this method to work as originally implemented in platform so I modelled the arguments on the other `perform` variant.

Reply:
Ah interesting, this looks good to me though :)